x/gov/keeper/msg_server.go

```diff
@@ -41,10 +41,6 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1.MsgSubmitPropos
 		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid proposer address: %s", err)
 	}
 
-	if err := validateDeposit(sdk.NewCoins(msg.InitialDeposit...)); err != nil {
```

Comment:
Rationale of deleting this check?

Reply:
Moved that check into `validateInitialDeposit` so we can have `validateDeposit` for deposits made after the proposal creation